dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/MemoryAllocation.java

```diff
@@ -87,7 +88,11 @@ private void createInitEvents(Program program) {
                 final String threadName = "Init_" + nextThreadId;
                 final Thread thread = new Thread(threadName, initThreadType, paramNames, nextThreadId,
                         EventFactory.newThreadStart(null));
-                thread.append(EventFactory.newInit(memObj, field));
+                if (program.getArch() == Arch.C11 || program.getArch() == Arch.OPENCL) {
+                    thread.append(EventFactory.newC11Init(memObj, field));
+                } else {
+                    thread.append(EventFactory.newInit(memObj, field));
+                }
```

Comment:
I think I would rather change `newC11Init(x,y)` by `newInit(x,y, boolean)`, implement `newCInit(x,y) = `newInit(x,y,false)` and avoid the if here.

Reply:
Will some other Archs also have unique tags for Init? Then this boolean might be not enough?

Reply:
What is this boolean supposed to mean? I would avoid adding any language specific stuff.

Reply:
> What is this boolean supposed to mean? I would avoid adding any language specific stuff.

I don't mean to the Init class, but to the factory. I don't see any difference between having `EventFactory.newInit(x, y, boolean)` and `EventFactory.newInit(x, y)` rather than `EventFactory.newC11Init(x, y)` and `EventFactory.newInit(x, y)` and what I propose simplifies the code (not only in the MemoryAllocation, but also in the factory where we avoid the duplicated code of this methods)

Reply:
I still don't know what the boolean is parameter is supposed to do? Is it about copying the feature tags from the `MemoryObject`? I think we should always do it since that is now a (nice) feature of the core language.
But `newC11Init` also adds the `C11.NONATOMIC` tag which is language specific and should not be part of the core construction.

As a general side note: Boolean parameters, especially one's that practically always take constant values (i.e., the caller passes `true/false`), tend to be hard to read on call sites unless it is an extremely natural parameter (e.g. `newInit(x, y, true)` is unclear but `newBoolConst(true)`  is natural). 
Fortunately, static factory methods have an advantage over constructors due to the fact that you can just provide two methods, one for `false` version and one for the `true` version, that tell you by name what is different.
From a different point of view, you can also say that `true/false` are just magic constants and those should be avoided. You either give them names on the call site like `static final bool INIT_WITH_C11_TAGS = true` or, well, you use two different functions in the factory.